src/Database/Dialect/PostgresDialectTrait.php

```diff
@@ -62,6 +65,69 @@ protected function _transformDistinct($query)
     }
 
     /**
+     * Apply translation steps to update queries.
+     *
+     * @param \Cake\Database\Query $query The query to translate
+     * @return \Cake\Database\Query The modified query
+     */
+    protected function _updateQueryTranslator($query)
+    {
+        $limit = $query->clause('limit');
+
+        if ($limit) {
+            $from = (new Query($query->connection()))
+                ->select('*')
+                ->from($query->clause('update'))
+                ->where($query->clause('where'))
+                ->limit($limit)
+                ->epilog('FOR UPDATE');
```

Comment:
I don't think there is a need for this, all queries are atomic, even if you use subqueries.